sql/rpl_gtid.cc

```diff
@@ -3343,6 +3344,7 @@ static my_bool report_audit_findings(void *entry, void *report_ctx_arg)
     }
 
     /* Report any out of order GTIDs */
+    if (report_ctx->report_out_of_order)
```

Comment:
Instead of stashing `late_gtids` until the end, `Binlog_gtid_state_validator::record` could instead warn/error out-of-order GTIDs itself live.